ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -432,11 +464,18 @@ private boolean isHostGroupBasedDeployment() {
     private void calculateTotalAgentsInHostgroups() {
         int agentsToExpect = 0;
         for (AmbariHostGroup hostGroup : getHostGroups()) {
-                agentsToExpect += hostGroup.getConfig(AmbariHostGroup.INITIAL_SIZE);
+            agentsToExpect += hostGroup.getConfig(AmbariHostGroup.INITIAL_SIZE);
+        }
+        if (getServer().agentOnServer()) {
+            agentsToExpect += Iterables.size(getAmbariServers());
         }
```

Comment:
Shouldn't it be something like this?
```
for (AmbariServer ambariServer: getAmbariServers()) {
    if (ambariServer.agentOnServer()) {
        agentsToExpect++;
    }
}
```

Instead of the `if` statement?